src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -280,14 +283,18 @@ private static string Decode(Stream stream, Encoding encoding, out Encoding actu
         {
             RoslynDebug.Assert(stream != null);
             RoslynDebug.Assert(encoding != null);
+            int length = 4096;
```

Comment:
>4096 [](start = 25, length = 4)

Perhaps add a `const int minBufferSize = 4096;` since the constant value is used a couple of times.